Vite 4 support for Quasar Vite Plugin (@quasar/vite-plugin)

### What happened?

Hi there ! 
It seems to be a dependency issue when creating a new quasar project when using the @quasar/vite-plugin. 
When scaffolding a new project using vite@latest, @vitejs/plugin-vue version is now 4.0.0.

During the installation procedure, the command ```npm install -D @quasar/vite-plugin``` fails due to a broken dependency. It will produce the following output :

``` bash
npm ERR! Could not resolve dependency:
npm ERR! peer @vitejs/plugin-vue@"^2.0.0 || ^3.0.0" from @quasar/vite-plugin@1.2.3
```



### What did you expect to happen?

I hope this dependency error will disapear soon :)

Thank you for this great framework and ecosystem.

### Reproduction URL

none

### How to reproduce?

1. Follow the project scaffolding procedure described in this page : ```https://quasar.dev/start/vite-plugin```

### Flavour

Vite Plugin (@quasar/vite-plugin)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite)

### Platforms/Browsers

Other

### Quasar info output

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

--- a/vite-plugin/package.json
+++ b/vite-plugin/package.json
@@ -32,9 +32,9 @@
     "rollup": "^2.58.0"
   },
   "peerDependencies": {
-    "@vitejs/plugin-vue": "^2.0.0 || ^3.0.0",
+    "@vitejs/plugin-vue": "^2.0.0 || ^3.0.0 || ^4.0.0",
     "quasar": "^2.8.0",
-    "vite": "^2.0.0 || ^3.0.0",
+    "vite": "^2.0.0 || ^3.0.0 || ^4.0.0",
     "vue": "^3.0.0"
   },
   "license": "MIT",

--- a/vite-plugin/src/plugin.js
+++ b/vite-plugin/src/plugin.js
@@ -17,7 +17,7 @@ function getConfigPlugin (opts) {
   return {
     name: 'vite:quasar:vite-conf',
 
-    config (viteConf) {
+    config (viteConf, { mode }) {
       const vueCfg = viteConf.plugins.find(entry => entry.name === 'vite:vue')
 
       if (vueCfg === void 0) {
@@ -25,7 +25,7 @@ function getConfigPlugin (opts) {
         process.exit(1)
       }
 
-      return getViteConfig(opts.runMode, viteConf)
+      return getViteConfig(opts.runMode, mode, viteConf)
     }
   }
 }

--- a/vite-plugin/src/vite-config.js
+++ b/vite-plugin/src/vite-config.js
@@ -1,6 +1,6 @@
 import { version } from 'quasar/package.json'
 
-export function getViteConfig (runMode, externalViteCfg) {
+export function getViteConfig (runMode, viteMode, externalViteCfg) {
   const viteCfg = {
     define: {
       __QUASAR_VERSION__: `'${ version }'`,
@@ -25,7 +25,7 @@ export function getViteConfig (runMode, externalViteCfg) {
   else {
     // Alias "quasar" package to its dev file (which has flags)
     // to reduce the number of HTTP requests while in DEV mode
-    if (externalViteCfg.mode === 'development') {
+    if (viteMode === 'development') {
       viteCfg.resolve = {
         alias: [
           { find: /^quasar$/, replacement: 'quasar/dist/quasar.esm.js' }

